Documentation/network/node-ipam.rst

```diff
@@ -15,11 +15,33 @@ to "advertise" the node's IPs directly inside a Service LoadBalancer. This featu
 is especially useful if you don't control the network you are running on and can't
 use either the L2 or BGP capabilities of Cilium.
 
-It works by getting the Node addresses of the selected pods and advertising them.
+It works by getting the Node addresses of the selected Nodes and advertising them.
 It will respect the ``.spec.ipFamilies`` to decide if IPv4 or IPv6 addresses
 shall be used and will use the ``ExternalIP`` addresses if any or the
 ``InternalIP`` addresses otherwise.
 
+If the Service has ``.spec.externalTrafficPolicy`` sets to ``Cluster`` Node IPAM
+will consider all your nodes as candidate for selection. Or else if
+``.spec.externalTrafficPolicy`` is set to ``Local``, Node IPAM will consider
+all the Pods selected by the Service (via their EndpointSlices) as candidate.
+
+.. warning::
+    Node IPAM will not work properly ``.spec.externalTrafficPolicy`` is set
+    to ``Local`` and that no EndpointSlice (or dummy EndpointSlice) are linked to
+    the corresponding Service.
+
+    As a result, you **cannot** use ``.spec.externalTrafficPolicy`` sets to ``Local``
+    with Cilium GatewayAPI or Cilium Ingress (although most GatewayAPI or Ingress
+    controller implementations should work/be compatible).
```

Comment:
```suggestion
    As a result, you **cannot** set ``.spec.externalTrafficPolicy`` to ``Local``
    with Cilium GatewayAPI or Cilium Ingress (although most GatewayAPI or Ingress
    controller implementations should work/be compatible).
```

`although most GatewayAPI or Ingress controller implementations should work/be compatible` -> Does this mean implementations other than Cilium? Why?

Reply:
Yep most implementation other than Cilium should work as Cilium implementation has a dummy endpoint (see this: https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/templates/cilium-ingress-service.yaml#L58) so we cannot see the actual Endpoints of Cilium envoy in there.

This was the trigger for me to look into this btw as before this PR Node IPAM would always do that so it was fully incompatible with Cilium ingress/GatewayAPI. Then I was kinda surprised this would work with other implementations of service LB and I looked into what is done with cloud providers etc.

Reply:
Ok, thanks for the details! It would be nice to make the comment more precise in that case, by mentioning that “implementations other than Cilium should work”, and maybe summarising where the limitation stems from.

Reply:
I tried to be a bit more more explicit there, let me know if this seems correct to you!